src/ca/mcgill/ecse211/lab4/LightLocalizer.java

```diff
@@ -0,0 +1,255 @@
+
+
+/**
+ * This clas implements light localization to complete the localization process for the robot. It
+ * detects a given amount of black lines on the test surface. Then, it uses the angles at which the
+ * robot detected those lines to orient itself towards the origin.
+ * 
+ * @author1 Cristian Ciungu
+ * @author2 Hao Shu
+ * @version 12-02-2019
+ * 
+ * 
+ */
+package ca.mcgill.ecse211.lab4;
+
+import lejos.hardware.Sound;
+import lejos.hardware.ev3.LocalEV3;
+import lejos.hardware.motor.EV3LargeRegulatedMotor;
+import lejos.hardware.sensor.EV3ColorSensor;
+import lejos.robotics.SampleProvider;
+import ca.mcgill.ecse211.odometer.*;
+import java.util.Arrays;
+
+public class LightLocalizer implements Runnable {

```

Comment:
No need to implement Runnable. You only need to do that when you have to do localization in parallel with other things that do not have their thread. Here, your localization routine does not run "in the background" like the odometer and display.